src/Cones/linmatrixineq.jl

```diff
@@ -129,18 +137,17 @@ function update_hess(cone::LinMatrixIneq)
 end
 
 function correction(cone::LinMatrixIneq, primal_dir::AbstractVector)
-    @assert cone.grad_updated
-    sumAinvAs = cone.sumAinvAs
+    @assert cone.feas_updated
     corr = cone.correction
-    dim = cone.dim
-
-    temp = zero(sumAinvAs[1])
-    temp .= 0
-    @inbounds for j in 1:dim, k in 1:dim
-        mul!(temp, sumAinvAs[j], sumAinvAs[k], primal_dir[j] * primal_dir[k], true)
-    end
-    @inbounds for i in 1:dim
-        corr[i] = real(dot(sumAinvAs[i], temp'))
+    As = cone.As
+    fact = cone.fact
+
+    dir_mat = sum(d_i * A_i for (d_i, A_i) in zip(primal_dir, As))
+    Y1 = fact \ dir_mat
+    Y2 = rdiv_sqrt!(Y1, fact)
+    M = Y2 * Y2'
```

Comment:
wrap with hermitian so that the dot below dispatches to `dot(::Hermitian, ::Hermitian)`